### DISCUSSION TITLE
About the cache implementation with `Date.now`

### DISCUSSION THREAD

**Discussion by EvandroLG** [TanStack/query#4936] [Ideas]
Hey folks! :wave:

I understand the reasons for using `Date.now` rather than `performance.now` for comparing times as it's way faster and has support for more browsers. It's said, I'd like to point out that it also comes with a clear vulnerability: being based on the system clock, any user can skip the cache and, depending on the application characteristics, it can be a real issue to use the library cache implementation.

So, I'd like to suggest changing a bit the current implementation by adding the possibility of using the `performance.now` method instead of the `Date.now`. Recently I worked on an application where I decided to create an in-house solution rather than using the library one and the reason was just because of the `Date.now` use.

Of course, I also wanted to hear what you all think about that.

**Comment by TkDodo** [comment]
not sure we can do this. According to [StackOverflow](https://stackoverflow.com/questions/30795525/performance-now-vs-date-now), they serve different purposes, mainly:

> performance.now() is relative to page load

this will likely break things between server-side rendering, or if data is persisted to e.g. localStorage and then restored later.
